calico_node/startup/startup.go

```diff
@@ -542,20 +560,44 @@ func ipv6Supported() bool {
 
 // createIPPool creates an IP pool using the specified CIDR.  This
 // method is a no-op if the pool already exists.
-func createIPPool(client *client.Client, cidr *net.IPNet, ipip bool) {
+func createIPPool(client *client.Client, cidr *net.IPNet, ipipMode ipip.Mode) {
 	version := cidr.Version()
 
-	log.Infof("Ensure default IPv%d pool is created. IPIP enabled: %t", version, ipip)
 	pool := &api.IPPool{
 		Metadata: api.IPPoolMetadata{
 			CIDR: *cidr,
 		},
 		Spec: api.IPPoolSpec{
 			NATOutgoing: true,
-			IPIP:        &api.IPIPConfiguration{Enabled: ipip},
+			IPIP: &api.IPIPConfiguration{
+				Enabled: ipipMode != ipip.Undefined,
+				Mode:    ipipMode,
+			},
 		},
 	}
 
+	// Use off when logging for disabled instead of blank
+	ipipModeString := "off"
+	if ipipMode != ipip.Undefined {
+		ipipModeString = string(ipipMode)
+	}
+
+	// Validate the pool Metadata and Spec
+	err := validator.Validate(pool.Metadata)
```

Comment:
I think the real fix is to make sure we validate the data in the API.  So I think we might have to bite the bullet and fix up that issue as it's a caring omission.